.pylintrc

```diff
@@ -0,0 +1,429 @@
+# paths.
+ignore=CVS,tensorflow_serving
+
+# Add files or directories matching the regex patterns to the blacklist.
+# The regex matches against base names, not paths.
+# Regex patterns can be comma(and newline)-separated
+ignore-patterns=
+    .*_pb2.py, # Ignore all files generated by the protocol buffer compiler
+
+# Pickle collected data for later comparisons.
+persistent=yes
+
+# List of plugins (as comma separated values of python modules names) to load,
+# usually to register additional checkers.
+load-plugins=
+
+# Use multiple processes to speed up Pylint.
+jobs=1
+
+# Allow loading of arbitrary C extensions. Extensions are imported into the
+# active Python interpreter and may run arbitrary code.
+unsafe-load-any-extension=no
+
+# A comma-separated list of package or module names from where C extensions may
+# be loaded. Extensions are loading into the active Python interpreter and may
+# run arbitrary code
+extension-pkg-whitelist=numpy
+
+# Allow optimization of some AST trees. This will activate a peephole AST
+# optimizer, which will apply various small optimizations. For instance, it can
+# be used to obtain the result of joining multiple strings with the addition
+# operator. Joining a lot of strings can lead to a maximum recursion error in
+# Pylint and this flag can prevent that. It has one side effect, the resulting
+# AST will be different than the one from reality. This option is deprecated
+# and it will be removed in Pylint 2.0.
+optimize-ast=no
+
+[MESSAGES CONTROL]
+
+# Only show warnings with the listed confidence levels. Leave empty to show
+# all. Valid levels: HIGH, INFERENCE, INFERENCE_FAILURE, UNDEFINED
+confidence=
+
+# Enable the message, report, category or checker with the given id(s). You can
+# either give multiple identifier separated by comma (,) or put this option
+# multiple time (only on the command line, not in the configuration file where
+# it should appear only once). See also the "--disable" option for examples.
+#enable=
+
+# Disable the message, report, category or checker with the given id(s). You
+# can either give multiple identifiers separated by comma (,) or put this
+# option multiple times (only on the command line, not in the configuration
+# file where it should appear only once).You can also use "--disable=all" to
+# disable everything first and then reenable specific checks. For example, if
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use"--disable=all --enable=classes
+# --disable=W"
+disable=
+    C0330, # Black disagrees with and explicitly violates this: https://github.com/python/black/issues/48
+    import-error, # Since we run Pylint before any of our builds in tox, this will always fail
+    wrong-import-order, # This check conflicts with flake8
+    no-else-return, # This check encourages code that is less readable
```

Comment:
just curious - is the rule saying

```
# Good
if x:
    return foo

return bar

# Bad

if x:
    return foo
else:
    return bar
```

and we're saying that the latter looks better?

Reply:
That is the correct interpretation of the rule. I wouldn't argue that the latter is better (although I personally find it more readable), but I don't see a reason to disallow it. Similar discussion here: https://github.com/SoCo/SoCo/issues/500